Add a search / jump to Pokémon function

Cycloneblaze opened the pull request:
Answers [this request](https://forums.gpx.plus/index.php?s=&showtopic=78892&view=findpost&p=2509804). As I noted there I figured it was inappropriate to leave a search function like this to Ctrl+F; right now that doesn't work, and to make it work we'd have to add text to the web page that's always visible with the name of each Pokémon. So instead I made a Javascript function using a search bar which searches in the alt text attribute of each image. We know we can use alt because we already use it to track the own/not owned state.

You can see how it works in the script, or I wrote a little usage guide in the getting started file, but two things to note. First, the search bar is actually added by Javascript, it's not in the HTML file(s). Instead, a new function at the top of the HTML file adds it in. It's not necessary to do it this way, but now if you don't have JS active or didn't download the JS file, the search bar won't appear at all, instead of it being there but not working.

Secondly, I added another function which adds a "tabindex" attribute to all Pokémon images. This is necessary to be able to focus on the images which is necessary for the search to highlight them (you can't normally tab to an <img>). But it also allows users with keyboards to tab through all the Pokémon and mark them with their keyboard. The search bar makes keyboard use a lot easier too; it's a bit tedious to tab through 400 Pokémon to get to Blaziken. The last thing necessary to make keyboard navigation possible is to make the mark function work on Enter as well, so I also did that. I'm quite happy with the incidental accessibility win here!

Finally, I've added a v4.2 branch as merge target so you can save this feature for the next versioned update of the tracker. 🙂

Cycloneblaze committed:
add :focus where there is :hover
Selecting Pokémon with the keyboard doesn't work right now, but when it does this will show the pop-up box on keyboard focus as well as mouse hover.

Cycloneblaze committed:
Add a search bar
It works as described in Getting Started, or the comments in the js file.

Cycloneblaze committed:
Improve result or failure visibility

Cycloneblaze committed:
Make marking work on press of Enter key
This completes keyboard navigation support. I think.

Cycloneblaze committed:
keydown -> keyup

Cycloneblaze committed:
Allow Enter in search box to search

Cycloneblaze committed:
Add a search hint

Level-X reviewed (approved):
Thank you Cyclone! ❤️

Cycloneblaze committed:
fix nav display on smaller screens
This necessitates changing the curent float-based layout to a grid layout with the nav as a sidebar, and a new HTML section to contain it.

Cycloneblaze commented:
Think this should work, we can make further tweaks in the v4.2 branch
